platform-ajax/src/test/java/com/softicar/platform/ajax/engine/AjaxDomEngineTest.java

```diff
@@ -11,10 +15,37 @@
 import com.softicar.platform.dom.styles.CssDisplay;
 import java.awt.Color;
 import java.awt.image.BufferedImage;
+import java.util.Optional;
 import org.junit.Test;
 
 public class AjaxDomEngineTest extends AbstractAjaxSeleniumLowLevelTest {
 
+	@Test
+	public void testSetAlternateResourceOnErrorWithPng() {
+
+		var png = openTestNode(() -> new DomImage(AjaxTestResources.TEST_PNG.getResource()));
+
+		// PNG remains PNG
+		var pngResource = assertOne(getResourceByHash(getAttributeValue(png, "src")));
+		assertEquals(MimeType.IMAGE_PNG, pngResource.getMimeType());
+		assertEquals("test.png", assertOne(pngResource.getFilename()));
+	}
+
+	/**
+	 * Most browsers do not support the TIFF image format, so we assert that it
+	 * was converted to PNG.
+	 */
+	@Test
+	public void testSetAlternateResourceOnErrorWithTiff() {
+
+		var tiff = openTestNode(() -> new DomImage(AjaxTestResources.TEST_TIFF.getResource()));
+
+		// TIFF converted to PNG
+		var tiffResource = assertOne(getResourceById(getAttributeValue(tiff, "src")));
+		assertEquals(MimeType.IMAGE_PNG, tiffResource.getMimeType());
+		assertEquals("test.tiff.png", assertOne(tiffResource.getFilename()));
```

Comment:
Shouldn't we assert the content of the output image - somewhere, and at least heuristically?
E.g. we could assert the dimensions of the output and the color of the pixel at `(0,0)`. That should be fairly straightforward, and it would be sufficient to ensure that we don't end up with an empty 1x1 image, with a black box, or something like this. `DomImageResourceConverterTest` would be an appropriate place for that.

Reply:
did so